src/app/api/search.api.ts

```diff
@@ -106,38 +107,17 @@ export class SearchApi {
         if(id === '') {
             return of([]);
         }
-        const GET_DATA: DocumentNode = gql`
-{
-  search {
-    query(query: "${id}", perPage: 10) {
-      nodes {
-        ... on Dataset {
-          id
-        }
-      }
-    }
-  }
-}`
 
-        /* eslint-disable  @typescript-eslint/no-explicit-any */
-        // @ts-ignore
-      return this.apollo.watchQuery({query: GET_DATA})
-            .valueChanges.pipe(map((result: ApolloQueryResult<any>) => {
-                if (result.data) {
-                    return SearchApi.searchValueAddToAutocomplete(result.data.search.query.nodes || [], id);
-                } else {
-                    return [];
-                }
-            }));
-    }
-    private static searchValueAddToAutocomplete(ngTypeaheadList: DatasetIDsInterface[], searchValue: string): DatasetIDsInterface[] {
-        const newArray: DatasetIDsInterface[] = JSON.parse(JSON.stringify(ngTypeaheadList));
-        if (searchValue) {
-            newArray.unshift({__typename: TypeNames.allDataType, id: searchValue});
-        }
-        return newArray;
+        return this.searchAutocompleteGQL.watch({query: id, perPage: 10}).valueChanges.pipe(map(result => {
```

Comment:
result should be with type

Reply:
Might be misunderstanding something, but isn't return value already typechecked as `Observable<DatasetIDsInterface[]>` as per function signature?

Reply:
Also my intention in this PR was to show an example of using a query generated from a template. I think ideally we would remove the hand-written `DatasetIDsInterface` type and return a type generated from the query as well.

This way any time we update the query - everything that depends on it will see the updated type.